### ISSUE TITLE
Train/Val split of Dogs&Cats Dataset

### ISSUE THREAD

**Issue by LayneH** [Z-T-WANG/NIPS2019DeepGamblers#1]
Hi,
Thank you for open-sourcing your code.

And I am a bit confusing about the train/val split in the dogs&cats experiment.
Do you use exactly the same split train/val split as Selective Net or randomly hold out 5000 samples as validation data?

**Comment by Z-T-WANG** [comment]
Thank you for your question.

We have used the last 5000 samples as the validation, or test set for reproducibility. 

Actually we do not know how the train/val split was done in the Selective Net paper. It is not shown in their released code, either.

**Comment by LayneH** [comment]
Thanks for your reply!

**Comment by Z-T-WANG** [comment]
Sorry, I just found that I have made a mistake. We actually did not use the last 5000 images as the validation data, and the train/val split was randomly. For reproducibility, we will give our random train/val split in the repository.

**Comment by LayneH** [comment]
Hi, I've just compared the performance of Deep Gambler with standard cross entropy on Cats&Dogs dataset.

Using the provided train/val split, I can reproduce the results as reported in your paper.
However, I find that CE yields surprisingly decent results, which are comparable with or even better than those reported in Deep Gambler / Selective Net. 

Here are the results:
```
CE (run #1):
Test    Coverage        Error
100,    100.00,         2.967
95,     94.87,          1.054
90,     90.10,          0.333
85,     85.03,          0.314
80,     79.90,          0.250

CE(run #2):
Test    Coverage        Error
100,    100.00,         2.767
95,     95.03,          1.228
90,     90.20,          0.517
85,     84.90,          0.393
80,     80.20,          0.333

CE (run #3)
Test    Coverage        Error
100,    100.00,         3.100
95,     95.13,          1.331
90,     90.10,          0.592
85,     84.97,          0.510
80,     80.10,          0.375
```

This is confusing. Do you have any idea?

**Comment by Z-T-WANG** [comment]
Actually the reason is simple. It is simply because when there are only two classes, entropy-based rejection is equivalent to SR, which is a strong strategy. It will not perform that well when there are 10 classes.

It seems that our cited data from Ref. [15] do not exactly match the accuracy of our trained models on Pytorch, and our models perform slightly better than theirs, even though we have tried our best to reproduce the same model architecture from their published codes. Nevertheless, we can still see the relative performance increase due to the abtention in our model is better than theirs.

**Comment by LayneH** [comment]
I think it would be better to conduct an ablation study that compare your method with CE on these 3 datasets using your codebase.
The following is my experiment result of CE on CIFAR10:
```
Test    Coverage        Error
100,    100.00,         5.663
95,     95.28,          3.779
90,     90.14,          2.122
85,     85.09,          1.175
80,     80.06,          0.765
75,     74.95,          0.484
70,     70.00,          0.339
```
The performance is also comparable with Deep Gambler.

**Comment by Z-T-WANG** [comment]
As you can see, your accuracy at 100 coverage is already above our 100 coverage accuracy, so the comparison would not be fair if you only look at the raw error rates. 

The model we use is not the vgg model provided by Pytorch, but we re-implemented it following the code of Ref. [15] to make a fair comparison as possible. In our tests, we found that CE-based abstention is always worse than SR, so we did not try it further.

**Comment by Z-T-WANG** [comment]
Since the issue is already solved through private communication, we close it for now.
